core/helpers/EEH_Sideloader.helper.php

```diff
@@ -210,91 +213,131 @@ public function get_new_file_name()
     /**
      * Downloads the file using the WordPress HTTP API.
      *
-     * @since 4.1.0
      * @return bool
+     * @since 4.1.0
      */
     public function sideload()
     {
-        // setup temp dir
-        $temp_file = wp_tempnam($this->_download_from);
-
-        if (!$temp_file) {
-            EE_Error::add_error(
-                esc_html__('Something went wrong with the upload.  Unable to create a tmp file for the uploaded file on the server', 'event_espresso'),
-                __FILE__,
-                __FUNCTION__,
-                __LINE__
-            );
-            return false;
-        }
+        try {
+            // setup temp dir
+            $temp_file = wp_tempnam($this->_download_from);
+
+            if (! $temp_file) {
+                throw new RuntimeException(
+                    esc_html__(
+                        'Something went wrong with the upload.  Unable to create a tmp file for the uploaded file on the server',
+                        'event_espresso'
+                    )
+                );
+            }
 
-        do_action('AHEE__EEH_Sideloader__sideload__before', $this, $temp_file);
+            do_action('AHEE__EEH_Sideloader__sideload__before', $this, $temp_file);
 
-        $wp_remote_args = apply_filters('FHEE__EEH_Sideloader__sideload__wp_remote_args', array( 'timeout' => 500, 'stream' => true, 'filename' => $temp_file ), $this, $temp_file);
+            $wp_remote_args = apply_filters(
+                'FHEE__EEH_Sideloader__sideload__wp_remote_args',
+                ['timeout' => 500, 'stream' => true, 'filename' => $temp_file],
+                $this,
+                $temp_file
+            );
 
-        $response = wp_safe_remote_get($this->_download_from, $wp_remote_args);
+            $response = wp_safe_remote_get($this->_download_from, $wp_remote_args);
 
-        if (is_wp_error($response) || 200 != wp_remote_retrieve_response_code($response)) {
-            unlink($temp_file);
-            if (defined('WP_DEBUG') && WP_DEBUG) {
-                EE_Error::add_error(
-                    sprintf(
-                        esc_html__('Unable to upload the file. Either the path given to download from is incorrect, or something else happened. Here is the path given: %s', 'event_espresso'),
-                        $this->_download_from
-                    ),
-                    __FILE__,
-                    __FUNCTION__,
-                    __LINE__
-                );
+            if ($this->isResponseError($response) || $this->isDownloadError($response) ) {
```

Comment:
extracted errors to these other files to reduce the cognitive load in this function
plus was able to reuse some of it